common/types/list.go

```diff
@@ -115,8 +118,16 @@ func (l *baseList) Contains(elem ref.Val) ref.Val {
 
 // ConvertToNative implements the ref.Val interface method.
 func (l *baseList) ConvertToNative(typeDesc reflect.Type) (interface{}, error) {
+	// TODO: Add support for conversion to Any
 	// JSON conversions are a special case since the 'native' type is a proto message.
-	if typeDesc == jsonValueType || typeDesc == jsonListValueType {
+	switch typeDesc {
+	case anyValueType:
+		json, err := l.ConvertToNative(jsonValueType)
```

Comment:
I think g.p.ListValue should be preferred instead of the less-specific g.p.Value.